core/kernel/shared/sprintf.rb

```diff
@@ -0,0 +1,945 @@
+        format("%E", -1.555555555).should == "-1.555556E+00"
+        format("%E", 1.444444444).should == "1.444444E+00"
+      end
+
+      it "displays Float::INFINITY as Inf" do
+        format("%E", Float::INFINITY).should == "Inf"
+        format("%E", -Float::INFINITY).should == "-Inf"
+      end
+
+      it "displays Float::NAN as NaN" do
+        format("%E", Float::NAN).should == "NaN"
+        format("%E", -Float::NAN).should == "NaN"
+      end
+    end
+
+    describe "f" do
+      it "converts floating point argument as [-]ddd.dddddd" do
+        format("%f", 10.952).should == "10.952000"
+        format("%f", -10.952).should == "-10.952000"
+      end
+
+      it "cuts excessive digits" do
+        format("%f", 1.123456789).should == "1.123457"
+      end
+
+      it "rounds the last significant digit to the closest one" do
+        format("%f", 1.555555555).should == "1.555556"
+        format("%f", -1.555555555).should == "-1.555556"
+        format("%f", 1.444444444).should == "1.444444"
+      end
+
+      it "displays Float::INFINITY as Inf" do
+        format("%f", Float::INFINITY).should == "Inf"
+        format("%f", -Float::INFINITY).should == "-Inf"
+      end
+
+      it "displays Float::NAN as NaN" do
+        format("%f", Float::NAN).should == "NaN"
+        format("%f", -Float::NAN).should == "NaN"
+      end
+    end
+
+    describe "g" do
+      context "the exponenta is less than -4" do
+        it "converts a floating point number using exponential form" do
+          format("%g", 0.0000123456).should == "1.23456e-05"
+          format("%g", -0.0000123456).should == "-1.23456e-05"
+
+          format("%g", 0.000000000123456).should == "1.23456e-10"
+          format("%g", -0.000000000123456).should == "-1.23456e-10"
+        end
+      end
+
+      context "the exponenta is greater than or equal to the precision (6 by default)" do
+        it "converts a floating point number using exponential form" do
+          format("%g", 1234567).should == "1.23457e+06"
+          format("%g", 1234567890123).should == "1.23457e+12"
+          format("%g", -1234567).should == "-1.23457e+06"
+        end
+      end
+
+      context "otherwise" do
+        it "converts a floating point number in dd.dddd form" do
+          format("%g", 0.0001).should == "0.0001"
+          format("%g", -0.0001).should == "-0.0001"
+          format("%g", 123456).should == "123456"
+          format("%g", -123456).should == "-123456"
+        end
+
+        it "cuts excessive digits in fractional part" do
+          format("%g", 12.12341111).should == "12.1234"
+          format("%g", -12.12341111).should == "-12.1234"
+        end
+
+        it "rounds the last significant digit to the closest one in fractional part" do
+          format("%g", 1.555555555).should == "1.55556"
+          format("%g", -1.555555555).should == "-1.55556"
+          format("%g", 1.444444444).should == "1.44444"
+        end
+
+        # strange behavior
```

Comment:
This looks OK to me, I think it tries to keep 6 digits shown.

Reply:
Agreed

Reply:
@eregon @marcandre 
my point is it isn't documented well and a little confusing as far as documentation declares

> Convert a floating point number ... in dd.dddd form ...

Having this description I would expect to have up to 4 digits after point

Reply:
Could you file it as a documentation issue then?